jdaviz/configs/default/plugins/plot_options/plot_options.py

```diff
@@ -625,9 +632,74 @@ def _update_stretch_histogram(self, msg={}):
             # in case only the sample has changed but its length has not,
             # we'll force the traitlet to trigger a change
             hist_mark.send_state('sample')
-
         self._stretch_histogram_needs_update = False
 
+    @observe('stretch_vmin_value', 'stretch_vmax_value', 'layer_selected',
+             'stretch_function_value', 'stretch_hist_nbins',
+             'image_contrast_value', 'image_bias_value')
+    def _update_stretch_curve(self, msg=None):
+        # Import here to prevent circular import (and not at the top of the method so the import
+        # check is avoided, whenever possible).
+        from jdaviz.configs.imviz.plugins.viewers import ImvizImageView
+        from jdaviz.configs.cubeviz.plugins.viewers import CubevizImageView
+
+        if (
+            not isinstance(self.viewer.selected_obj, (ImvizImageView, CubevizImageView)) or
+            not hasattr(self, 'stretch_histogram')
+        ):
+            # don't update histogram if selected viewer is not an image viewer,
+            # or the stretch histogram hasn't been initialized:
+            return
+
+        mark_label_prefix = "stretch_curve: "
+        for layer in self.layer.selected_obj:
+            # clear old mark, if it exists:
+            mark_label = f'{mark_label_prefix}{layer.label}'
+            mark_exists = mark_label in self.stretch_histogram.marks
+
+            # create the new/updated mark following the colormapping
+            # procedure in glue's CompositeArray:
+            layer_state = layer.state.as_dict()
+            interval = ManualInterval(self.stretch_vmin.value, self.stretch_vmax.value)
+            contrast_bias = ContrastBiasStretch(layer_state['contrast'], layer_state['bias'])
+            stretch = stretches.members[layer_state['stretch']]
+            layer_cmap = layer_state['cmap']
+
+            # create a photoshop style "curve" for the stretch function
+            curve_x = np.linspace(self.stretch_vmin.value, self.stretch_vmax.value, 50)
+            curve_y = interval(curve_x, clip=False)
+            curve_y = contrast_bias(curve_y, out=curve_y, clip=False)
+            curve_y = stretch(curve_y, out=curve_y, clip=False)
+            curve_y = layer_cmap(curve_y)[:, 0]
+
+            if not mark_exists:

```

Comment:
Would it make more sense to have this after the following for loop, so we're not immediately checking a mark we just created that already has the right x and y?

Reply:
Good idea, done!